src/asl.ts

```diff
@@ -2392,10 +2441,25 @@ export class ASL {
           return {
             Or: [localToCondition(expr.left), localToCondition(expr.right)],
           };
-        } else if (expr.op === "+" || expr.op === "-") {
-          throw new Error(
-            `operation '${expr.op}' is not supported in a Condition`
-          );
+        } else if (expr.op === "??" || expr.op === "=") {
+          // FIXME: the potentially circular reference here (eval -> toCondition -> eval -> toCondition) is not good. Should toCondition and eval also be merged?
```

Comment:
Why would it recurse infinitely?

Reply:
Not right now, but its fragile. I think the solution is to reflect condition as an alternative to JsonPath and LiteralValue, merging `toCondition` with `eval` and then locally convert condition to `jsonPath` or `Condition` based on the need.